RecoTauTag/RecoTau/plugins/PFRecoTauDiscriminationByIsolation.cc

```diff
@@ -195,10 +228,29 @@ class PFRecoTauDiscriminationByIsolation : public PFTauDiscriminationProducerBas
   double offsetRelativeSumPt_;
   double customIsoCone_;
   
+  bool applyPhotonPtSumOutsideSignalConeCut_;
+  double maxAbsPhotonSumPt_outsideSignalCone_;
+  double maxRelPhotonSumPt_outsideSignalCone_;
+
+  bool applyFootprintCorrection_;
+  struct footprintCorrectionType
+  {
+    footprintCorrectionType(const std::string& selection, const std::string& offset)
+      : selection_(selection),
+	offset_(offset)
+    {}
+    ~footprintCorrectionType() {}
+    StringCutObjectSelector<PFTau> selection_;
+    StringObjectFunction<PFTau> offset_;
+  };
+  std::vector<footprintCorrectionType*> footprintCorrections_;
```

Comment:
Hi Slava,

Now it's throwing the following error:

/afs/cern.ch/work/a/ajohnson/CMSSW_7_6_X_2015-08-04-2300/src/RecoTauTag/RecoTau/plugins/PFRecoTauDiscriminationByIsolation.cc:191:51: error: expected primary-expression before 'it'
       delete std::unique_ptr<FootprintCorrection> it;

in the destructor. Previously it had just been "delete (*it)" 
I wasn't sure whether to use the line:
delete std::unique_ptr<FootprintCorrection> it;
or
delete std::vector<std::unique_ptr<FootprintCorrection> > it;
In any case, either throw compiler errors